sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java

```diff
@@ -1262,6 +1262,87 @@ public AzureKeyCredential getCredential() {
      */
     Flux<FeedResponse<User>> queryUsers(String databaseLink, SqlQuerySpec querySpec, CosmosQueryRequestOptions options);
 
+    /**
+     * Reads a client encryption key.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single resource response with the read client encryption key.
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param clientEncryptionKeyLink the client encryption key link.
+     * @param options  the request options.
+     * @return a {@link Mono} containing the single resource response with the read user or an error.
+     */
+    Mono<ResourceResponse<ClientEncryptionKey>> readClientEncryptionKey(String clientEncryptionKeyLink, RequestOptions options);
+
+    /**
+     * Creates a client encryption key.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single resource response with the created client encryption key.
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param databaseLink the database link.
+     * @param clientEncryptionKey the client encryption key to create.
+     * @param options      the request options.
+     * @return a {@link Mono} containing the single resource response with the created client encryption key or an error.
+     */
+    Mono<ResourceResponse<ClientEncryptionKey>> createClientEncryptionKey(String databaseLink, ClientEncryptionKey clientEncryptionKey, RequestOptions options);
+
+    /**
+     * Replaces a client encryption key.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single resource response with the replaced client encryption key.
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param clientEncryptionKey    the client encryption key to use.
+     * @param options the request options.
+     * @return a {@link Mono} containing the single resource response with the replaced client encryption keyer or an error.
+     */
+    Mono<ResourceResponse<ClientEncryptionKey>> replaceClientEncryptionKey(ClientEncryptionKey clientEncryptionKey, String nameBasedLink, RequestOptions options);
```

Comment:
>String [](start = 116, length = 6)

doesn't clientEncryptionKey have the nameBasedLink within it?

Reply:
We have nameBasedLink in CosmosAsyncClientEncryptionKey , which is calling this method. We follow same for other resources like container/database